src/guide/composition-api-introduction.md

```diff
@@ -76,7 +76,7 @@ Now that we know the **why** we can get to the **how**. To start working with th
 The new `setup` component option is executed **before** the component is created, once the `props` are resolved, and serves as the entry point for composition API's.
 
 ::: warning
-Because the component instance is not yet created when `setup` is executed, there is no `this` inside a `setup` option. This means, with the exception of `props`, you won't be able to access any properties declared in the component – **local state**, **computed properties** or **methods**.
+Although the component instance has been created, it is not bound to `this` when the setup is executed, so `this` cannot be used in the setup, and the processing of props is also completed later, this means, with the exception of `props`, you won't be able to access any properties declared in the component – **local state**, **computed properties** or **methods**.
```

Comment:
There's quite a lot goes on internally and it's difficult to document accurately without causing confusion,

The thing most people would call the 'component instance' is the proxy that exposes the `data` properties, `methods`, etc.. That's stored in the `proxy` property of the `ComponentInternalInstance`. That does exist at the point `setup` is called and is accessible via `getCurrentInstance().proxy`, though that isn't something we want to be encouraging.

Whether the component instance is 'created' depends on how exactly you define *created*. The proxy object does exist but we're still in the process of populating it. The `created` lifecycle hook isn't called until after the object is fully populated and ready to render.

I would agree that it is a little misleading to imply that the `this` value in `setup` can't reference the component instance because it doesn't exist yet. That object does exist, so it definitely could be the `this` value. It's more of a design decision not to expose that object to prevent it being manipulated in `setup`.

My suggestion would be to avoid mentioning whether the component instance exists altogether and instead focus on the practical considerations. I believe the purpose of this warning is to mention that `this` is not available and that the things that are usually exposed via `this` won't exist yet. So perhaps let's just say that more directly?

```suggestion
You should avoid using `this` inside `setup` as it won't refer to the component instance. `setup` is called before `data` properties, `computed` properties or `methods` are resolved, so they won't be available within `setup`.
```

In addition, what do you think about moving this warning down a bit, so that it's after the next paragraph? The warning is important but I think we need to explain that `setup` is a function before we mention the unavailability of `this`.